includes/distributor-customizations/class-author-distribution.php

```diff
@@ -0,0 +1,182 @@
+ * On the target site, the plugin will create the authors if they don't exist, and override the byline
+ */
+class Author_Distribution {
+
+	/**
+	 * Initializes the class
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		add_filter( 'dt_push_post_args', [ __CLASS__, 'add_author_data_to_push' ], 10, 2 );
+		add_filter( 'rest_prepare_post', [ __CLASS__, 'add_author_data_to_pull' ], 10, 3 );
+	}
+
+	/**
+	 * Filters the post data sent on a push to add the author data.
+	 *
+	 * @param array   $post_body The post data.
+	 * @param WP_Post $post The post object.
+	 * @return array
+	 */
+	public static function add_author_data_to_push( $post_body, $post ) {
+		$authors = self::get_authors_for_distribution( $post );
+		if ( ! empty( $authors ) ) {
+			$post_body['newspack_network_authors'] = $authors;
+		}
+		return $post_body;
+	}
+
+	/**
+	 * Filters the post data for a REST API response.
+	 *
+	 * This acts on requests made to pull a post from this site.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WP_Post          $post     Post object.
+	 * @param WP_REST_Request  $request  Request object.
+	 */
+	public static function add_author_data_to_pull( $response, $post, $request ) {
+		if (
+			empty( $request->get_param( 'distributor_request' ) ) ||
+			'GET' !== $request->get_method() ||
+			'edit' !== $request->get_param( 'context' ) ||
+			empty( $request->get_param( 'id' ) )
+		) {
+			return $response;
+		}
+
+		$authors = self::get_authors_for_distribution( $post );
+
+		if ( ! empty( $authors ) ) {
+			$data                             = $response->get_data();
+			$data['newspack_network_authors'] = $authors;
+			$response->set_data( $data );
+		}
+
+		return $response;
+
+	}
+
+	/**
+	 * Get the authors of a post to be added to the distribution payload.
+	 *
+	 * @param WP_Post $post The post object.
+	 * @return array An array of authors.
+	 */
+	public static function get_authors_for_distribution( $post ) {
+		if ( ! function_exists( 'get_coauthors' ) ) {
+			return [ self::get_wp_user_for_distribution( $post->post_author ) ];
+		}
+
+		$co_authors = get_coauthors( $post->ID );
+		if ( empty( $co_authors ) ) {
+			return [ self::get_wp_user_for_distribution( $post->post_author ) ];
+		}
+
+		$authors = [];
+
+		foreach ( $co_authors as $co_author ) {
+			if ( is_a( $co_author, 'WP_User' ) ) {
+				$authors[] = self::get_wp_user_for_distribution( $co_author );
+			}
+			$authors[] = self::get_guest_author_for_distribution( $co_author );
+		}
+
+		return $authors;
+
+	}
+
+	/**
+	 * Gets the user data of a WP user to be distributed along with the post.
+	 *
+	 * @param int|WP_Post $user The user ID or object.
+	 * @return array
+	 */
+	public static function get_wp_user_for_distribution( $user ) {
+		if ( ! is_a( $user, 'WP_User' ) ) {
+			$user = get_user_by( 'ID', $user );
+		}
+
+		if ( ! $user ) {
+			return [];
```

Comment:
Mainly because all this happens as side effects in the background and we don't have any opportunity to surface these errors to the user. How would you debug it?

But yes, I guess we could return a wp_error there... but then we need more lines in the method that call this to check what has been returned

Reply:
Not really seeing a benefit of returning a WP_Error, but added some debugging on https://github.com/Automattic/newspack-network/pull/17/commits/2e26d030e327756f0916fbf017564cfb639dbe82